src/compressor.js

```diff
@@ -0,0 +1,76 @@
+define(function (require) {
+	'use strict';
+
+	var p5sound = require('master');
+	var Effect = require('effect');
+  var CustomError = require('errorHandler');
+
+	p5.Compressor = function() {
+		Effect.call(this);
+
+		this.compressor = this.ac.createDynamicsCompressor();
+    this.input.connect(this.compressor);
+    this.compressor.connect(this.wet);
+	};
+
+	p5.Compressor.prototype = Object.create(Effect.prototype);
+
+	p5.Compressor.prototype.process = function(src, attack, knee, 
+                                      ratio, threshold, release) {
+		src.connect(this.input);
+
+		this.set(attack, knee, ratio, threshold, release);
+	};
+
+
+  p5.Compressor.prototype.set = function (attack, knee, 
+                                ratio, threshold, release) {
+
+    // var self = this;
+    // var errorTrace = new Error().stack;
+
+    if (attack) {this.attack(attack);}
```

Comment:
for all of these, `if (attack)` is a == comparison, and `0` will be "falsey". So if we want to allow for `0`, we should do a more explicit check like `if (attack) !== 'undefined'`